### ISSUE TITLE
trimOnSave on new version vscode doesn't work

### ISSUE THREAD

**Issue by gggin** [shardulm94/vscode-trailingspaces#29]
Version: 1.33.0
Commit: 0dd516dd412d42323fc3464531b1c715d51c4c1a
Date: 2019-04-04T15:08:57.775Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Linux x64 4.15.0-43-generic

only on config:
"trailing-spaces.trimOnSave": true,

**Comment by shardulm94** [comment]
I am not able to reproduce this. Can you provide more information?
Does highlighting work? Is there any status bar message you see? If you go to `Help > Toggle Developer Tools` is there any error message you see?

**Comment by gggin** [comment]
Ok, I found that if I save file on my own the trimOnSave is ok. but I use this 
```
"files.autoSave": "afterDelay",
  "files.autoSaveDelay": 2000,

  "trailing-spaces.trimOnSave": true,
  "trailing-spaces.showStatusBarMessage": true,
```
Origin version vscode is good, vscode autosave will trigger trimOnSave, but this version fails.

**Comment by shardulm94** [comment]
Ohk... yes that makes sense. In v0.3.0, we changed to a newer VSCode API will allows the extension to apply pre-save edits rather than saving files twice.
https://github.com/shardulm94/vscode-trailingspaces/blob/master/src/trailing-spaces/loader.ts#L87
However, I only check for saves which are triggered manually and not automatically. I guess to be backwards compatible with older versions, I can enable this functionality for all types of saves.

**Comment by shardulm94** [comment]
Fixed in fbf81ba534b57d2ebbc8dfaf46ad69f90cf9aa43
Can you check if the latest beta release fixes this?
https://github.com/shardulm94/vscode-trailingspaces/releases/tag/v0.3.1-rc1
https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix

**Comment by gggin** [comment]
it's fixed. Thank you very much.
